eras/babbage/impl/src/Cardano/Ledger/Babbage/Rules/Utxo.hs

```diff
@@ -242,6 +242,80 @@ validateCollateralEqBalance bal txcoll =
     SNothing -> pure ()
     SJust tc -> failureUnless (bal == tc) (UnequalCollateralReturn bal tc)
 
+-- > getValue txout ≥ inject ( ⌈ serSize txout ∗ coinsPerUTxOWord pp / 8 ⌉ )
+validateOutputTooSmallUTxO ::
+  ( Era era,
+    ToCBOR (Core.Value era),
+    HasField "_coinsPerUTxOWord" (Core.PParams era) Coin
+  ) =>
+  Core.PParams era ->
+  [Core.TxOut era] ->
+  Test (UtxoPredicateFailure era)
+validateOutputTooSmallUTxO pp outs = failureUnless (null outputsTooSmall) $ OutputTooSmallUTxO outputsTooSmall
+  where
+    Coin coinsPerUTxOWord = getField @"_coinsPerUTxOWord" pp
+    serSize = fromIntegral . BSL.length . serialize . getField @"value"
+    outputsTooSmall =
+      filter
+        ( \out ->
+            let v = getField @"value" out
+             in -- pointwise is used because non-ada amounts must be >= 0 too
+                not $ Val.pointwise (>=) v (Val.inject $ Coin (serSize out * coinsPerUTxOWord `div` 8))
+        )
+        outs
+
+-- > serSize (getValue txout) ≤ maxValSize pp
+validateOutputTooBigUTxO ::
+  ( HasField "_maxValSize" (Core.PParams era) Natural,
+    HasField "value" (Core.TxOut era) (Core.Value era),
+    ToCBOR (Core.Value era)
+  ) =>
+  Core.PParams era ->
+  [Core.TxOut era] ->
+  Test (UtxoPredicateFailure era)
+validateOutputTooBigUTxO pp outs =
+  failureUnless (null outputsTooBig) $ OutputTooBigUTxO outputsTooBig
+  where
+    maxValSize = getField @"_maxValSize" pp
+    outputsTooBig = foldl' accum [] outs
+    accum ans out =
+      let v = getField @"value" out
+          serSize = fromIntegral $ BSL.length $ serialize v
+       in if serSize > maxValSize
+            then (fromIntegral serSize, fromIntegral maxValSize, out) : ans
+            else ans
+
+-- > a ∈ Addr_bootstrap ⇒ bootstrapAttrsSize a ≤ 64
+validateOutputBootAddrAttrsTooBig ::
+  Era era =>
+  [Core.TxOut era] ->
+  Test (UtxoPredicateFailure era)
+validateOutputBootAddrAttrsTooBig outs =
+  failureUnless (null outputsAttrsTooBig) $ OutputBootAddrAttrsTooBig outputsAttrsTooBig
+  where
+    outputsAttrsTooBig =
+      filter
+        ( \txOut ->
+            case getTxOutBootstrapAddress txOut of
+              Just addr -> bootstrapAddressAttrsSize addr > 64
+              _ -> False
+        )
+        outs
+
+-- > ∀(_ → (a, _)) ∈ allOuts txb, netId a = NetworkId
+validateWrongNetwork ::
```

Comment:
I think we could alternatively change the old version of this function to take outputs instead of a tx body, and then re-use it.